Add beta 45 analyses

poiru opened the pull request:
The analyses are the same for the most part with the following exceptions:
- `e10s_crash_rate.ipynb` is completely different. It now calculates crashes per 1000 usage hours per build instead of crashes per day.
- `e10s_experiment.ipynb` is mostly the same, but the BHR stuff (`add_gecko_activity`) has been changed to account for child payloads. When there are multiple child payloads in a given ping, the analysis uses the average hangs per minute.

vitillo commented:
nit: rename `get_hangs_per_mintute` to `get_hangs_per_minute`

vitillo commented:
In cell 13 we should consider the sum of all child hangs, not the average.

vitillo commented:
Some histograms look weird, e.g. MEMORY_HEAP_ALLOCATED, any idea why?

vitillo commented:
You could sample down the number of clients by say 2x to make the analysis faster; I don't see a strong reason to have that many clients in each branch.

vitillo commented:
Since we are going to run two experiments on beta45 could you create a less generic folder for this one?

vitillo commented:
The crash stat analysis seems weird to me, the length of the subsessions should be more or less equivalent between the branches of the experiment, but they are not. Could you print the number of submissions received for the control and experiment branch?

poiru commented:
> Some histograms look weird, e.g. MEMORY_HEAP_ALLOCATED, any idea why? I have seen negative values have a similar effect on plots in the past.

Yep, I think it's malformed data in some ping(s). When I first did the analysis, I used a smaller subset and it didn't have this issue.

vitillo commented:
Please add INPUT_EVENT_RESPONSE_MS (see Bug 1223780)

poiru commented:
@vitillo Updated PR. Added crash analysis with/without addons as well.

poiru commented:
@vitillo `e10s_crash_rate_without_addons.ipynb` doesn't use the same subset, let me fix that. But comparing e10s_crash_rate.ipynb (all profiles) and e10s_crash_rate_with_addons.ipynb, we can see that the vast majority of pings have addons. I'm not sure if this intentional or not.

vitillo commented:
Updated with what exactly? Please update this PR only with changes that reflect the comment and file and file another PR for new stuff.

poiru commented:
>Updated with what exactly? Please update this PR only with changes that reflect the comments and file another PR for new stuff.

Removed the new stuff.

>nit: rename get_hangs_per_mintute to get_hangs_per_minute

Fixed.

>Some histograms look weird, e.g. MEMORY_HEAP_ALLOCATED, any idea why? I have seen negative values have a similar effect on plots in the past.

I didn't do anything about this since its caused by to malfored histograms. Do you think we should validate the histgorams in this PR or leave it for another one?

>The crash stat analysis seems weird to me, the length of the subsessions should be more or less equivalent between the branches of the experiment, but they are not. Could you print the number of submissions received for the control and experiment branch?

Done.

vitillo commented:
The crash stats analysis looks better. The experiment branch has about 10% less clients than the control one, probably because e10s was force-disabled for some clients. Without a confidence interval or error bar for the normalized number of crashes it's hard to compare the numbers though. Few buggy clients could drive up the total total number of crashes, for instance. This is something you might want to think about for a future refinement of the analysis.

vitillo commented:
> I didn't do anything about this since its caused by to malfored histograms. Do you think we should validate the histgorams in this PR or leave it for another one?

This should get fixed in this PR.

poiru commented:
@vitillo I updated `e10s_experiment.ipynb` to use the new derived dataset (with `sampleId <= 6`), which got rid of the offending ping(s). I'll look into malformed histogram, but perhaps this can be merged now?

vitillo commented:
`e10s_top_hang_stacks.ipynb` is not using the latest dataset.

poiru commented:
@vitillo Updated `e10s_top_hang_stacks.ipynb` to use the same dataset as the others. Sorry for the delay!

vitillo commented:
Looks good, thanks.
